l10n_br_account_product/models/account_invoice.py

```diff
@@ -37,6 +37,7 @@
 
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
+    _order = "internal_number desc, date_hour_invoice desc"
```

Comment:
To correct order you must invert the order of columns: _order = "date_hour_invoice desc, internal_number desc"